web/components/scroll-to-top-button.tsx

```diff
@@ -0,0 +1,48 @@
+import { ArrowUpIcon } from '@heroicons/react/solid'
+import clsx from 'clsx'
+import { useEffect, useState } from 'react'
+import { Row } from './layout/row'
+
+export function ScrollToTopButton(props: { className?: string }) {
+  const { className } = props
+  const [visible, setVisible] = useState(false)
+
+  const onScroll = () => {
```

Comment:
I think I'd move this function inside the `useEffect`. One reason is it creates a new function each render here, while only the first is being referenced in the effect.